insalan/cms/models.py

```diff
@@ -58,6 +58,28 @@ class Meta:
     def __str__(self) -> str:
         return f"[Content] {self.name}"
 
+    def save(self, *args, **kwargs):
+        """
+        Override the save method to ensure that the content is valid.
+        """
+        # Apply some operations on the content to make it look better
+        if self.name == "planning":
+            if self.content.startswith("<meta"):
+                self.content = "\n".join(self.content.split("\n")[1:])
+            self.content = self.content.replace(
+                "background-color:#f3f3f3", 
+                "background-color:#434343"
+            )
+            self.content = self.content.replace("color:#555555", "color:#000000")
+            self.content = self.content.replace(";color:#434343;", ";color:#f3f3f3;")
+            self.content = self.content.replace("background-color:#ffffff;", "")
+            self.content = re.sub(
+                r"border-right:1px SOLID #[a-f0-9]*",
+                "border-right:1px SOLID #000000",
+                self.content
+            )
+        super().save(*args, **kwargs)
+
```

Comment:
Why wouldn't that kind of things make more sense in the frontend's CSS and just set some classes here ?

Reply:
The aim here is to make it as easy as possible to add a schedule. The idea is to start by exporting the schedule and then to make it readable on the site. The problem is that an exported google spreadsheet already contains css. So you have to change the colours used in this CSS, which could be done on the client side but which is simpler if done only once during registration.

You can find a test version to look at the export here : 
```
<meta http-equiv="Content-Type" content="text/html; charset=utf-8"><link type="text/css" rel="stylesheet" href="resources/sheet.css" >
<style type="text/css">.ritz .waffle a { color: inherit; }.ritz .waffle .s31{border-bottom:1px SOLID #000000;background-color:#ffffff;}.ritz .waffle .s35{border-bottom:1px SOLID #000000;border-right:1px SOLID #000000;background-color:#f3f3f3;text-align:center;color:#555555;font-family:'docs-Roboto',Arial;font-size:14pt;vertical-align:middle;white-space:normal;overflow:hidden;direction:ltr;padding:2px 3px 2px 3px;}.ritz .waffle .s36{border-bottom:1px SOLID #000000;border-right:1px SOLID #000000;background-color:#c9daf8;text-align:center;color:#555555;font-family:'docs-Roboto',Arial;font-size:10pt;vertical-align:middle;white-space:normal;overflow:hidden;direction:ltr;padding:2px 3px 2px 3px;}.ritz .waffle .s19{border-bottom:1px SOLID #000000;border-right:1px SOLID #000000;background-color:#87f59c;text-align:center;color:#555555;font-family:'docs-Roboto',Arial;font-size:17pt;vertical-align:middle;white-space:normal;overflow:hidden;direction:ltr;padding:2px 3px 2px 3px;}.ritz .waffle .s28{border-right:1px SOLID #000000;background-color:#ffffff;}.ritz .waffle .s41{border-bottom:1px SOLID #000000;border-right:1px SOLID #000000;background-color:#a8ffd5;text-align:center;color:#555555;font-family:'docs-Roboto',Arial;font-size:10pt;vertical-align:middle;white-space:normal;overflow:hidden;direction:ltr;padding:2px 3px 2px 3px;}.ritz .waffle .s46{background-color:#ffffff;text-align:left;color:#555555;font-family:'docs-Roboto',Arial;font-size:10pt;vertical-align:middle;white-space:nowrap;direction:ltr;padding:2px 3px 2px 3px;}.ritz .waffle .s38{border-right:1px SOLID #000000;background-color:#ffffff;text-align:left;color:#555555;font-family:'docs-Roboto',Arial;font-size:10pt;vertical-align:middle;white-space:normal;overflow:hidden;direction:ltr;padding:2px 3px 2px 3px;}.ritz .waffle .s23{border-bottom:1px SOLID #000000;border-right:1px SOLID #000000;background-color:#f3f3f3;text-align:center;color:#555555;font-family:'docs-Roboto',Arial;font-size:12pt;vertical-align:middle;white-space:normal;overflow:hidden;direction:ltr;padding:2px 3px 2px 3px;}.ritz .waffle .s37{background-color:#ffffff;text-align:left;color:#555555;font-family:'docs-Roboto',Arial;font-size:10pt;vertical-align:middle;white-space:normal;overflow:hidden;direction:ltr;padding:2px 3px 2px 3px;}.ritz .waffle .s27{border-bottom:1px SOLID #f3f3f3;background-color:#ffffff;text-align:right;color:#434343;font-family:'docs-Roboto',Arial;font-size:10pt;vertical-align:middle;white-space:normal;overflow:hidden;direction:ltr;padding:2px 3px 2px 3px;}.ritz .waffle .s4{border-right:1px SOLID #000000;background-color:#ffffff;text-align:center;font-weight:bold;color:#d9d9d9;font-family:'docs-Roboto',Arial;font-size:19pt;vertical-align:bottom;white-space:nowrap;direction:ltr;padding:2px 3px 2px 3px;}.ritz .waffle .s14{border-bottom:1px SOLID #f3f3f3;border-right:1px SOLID #d9d9d9;background-color:#ffffff;text-align:right;color:#434343;font-family:'docs-Roboto',Arial;font-size:10pt;vertical-align:middle;white-space:normal;overflow:hidden;direction:ltr;padding:2px 3px 2px 3px;}.ritz .waffle .s17{border-bottom:1px SOLID #000000;border-right:1px SOLID #000000;background-color:#d9d2e9;text-align:center;color:#555555;font-family:'docs-Roboto',Arial;font-size:17pt;vertical-align:middle;white-space:normal;overflow:hidden;direction:ltr;padding:2px 3px 2px 3px;}.ritz .waffle .s33{border-bottom:1px SOLID #000000;background-color:#ffffff;text-align:center;color:#555555;font-family:'docs-Roboto',Arial;font-size:17pt;vertical-align:middle;white-space:normal;overflow:hidden;direction:ltr;padding:2px 3px 2px 3px;}.ritz .waffle .s43{border-bottom:1px SOLID #000000;border-right:1px SOLID #000000;background-color:#f3f3f3;text-align:right;color:#434343;font-family:'docs-Roboto',Arial;font-size:10pt;vertical-align:middle;white-space:normal;overflow:hidden;direction:ltr;padding:2px 3px 2px 3px;}.ritz .waffle .s40{border-bottom:1px SOLID #000000;border-right:1px SOLID #000000;background-color:#ffffff;text-align:left;color:#555555;font-family:'docs-Roboto',Arial;font-size:10pt;vertical-align:middle;white-space:normal;overflow:hidden;direction:ltr;padding:2px 3px 2px 3px;}.ritz .waffle .s25{border-bottom:1px SOLID #000000;border-right:1px SOLID #000000;background-color:#ead1dc;text-align:center;color:#555555;font-family:'docs-Roboto',Arial;font-size:11pt;vertical-align:middle;white-space:normal;overflow:hidden;direction:ltr;padding:2px 3px 2px 3px;}.ritz .waffle .s11{border-right:1px SOLID #000000;background-color:#ffffff;text-align:left;color:#555555;font-family:'docs-Roboto',Arial;font-size:10pt;vertical-align:middle;white-space:nowrap;direction:ltr;padding:2px 3px 2px 3px;}.ritz .waffle .s20{border-bottom:1px SOLID #000000;border-right:1px SOLID #000000;background-color:#fab8ae;text-align:center;color:#555555;font-family:'docs-Roboto',Arial;font-size:17pt;vertical-align:middle;white-space:normal;overflow:hidden;direction:ltr;padding:2px 3px 2px 3px;}.ritz .waffle .s9{border-bottom:1px SOLID #f3f3f3;border-right:1px SOLID #d9d9d9;background-color:#f3f3f3;text-align:right;color:#434343;font-family:'docs-Roboto',Arial;font-size:10pt;vertical-align:middle;white-space:normal;overflow:hidden;direction:ltr;padding:2px 3px 2px 3px;}.ritz .waffle .s18{border-bottom:1px SOLID #000000;border-right:1px SOLID #000000;background-color:#f9cb9c;text-align:center;color:#555555;font-family:'docs-Roboto',Arial;font-size:17pt;vertical-align:middle;white-space:normal;overflow:hidden;direction:ltr;padding:2px 3px 2px 3px;}.ritz .waffle .s7{border-bottom:1px SOLID #000000;border-right:1px SOLID #000000;background-color:#ffffff;text-align:center;font-weight:bold;color:#d9d9d9;font-family:'docs-Roboto',Arial;font-size:19pt;vertical-align:bottom;white-space:nowrap;direction:ltr;padding:2px 3px 2px 3px;}.ritz .waffle .s29{border-bottom:1px SOLID #f3f3f3;background-color:#f3f3f3;text-align:right;color:#434343;font-family:'docs-Roboto',Arial;font-size:10pt;vertical-align:middle;white-space:normal;overflow:hidden;direction:ltr;padding:2px 3px 2px 3px;}.ritz .waffle .s3{background-color:#ffffff;text-align:center;font-weight:bold;color:#d9d9d9;font-family:'docs-Roboto',Arial;font-size:19pt;vertical-align:bottom;white-space:nowrap;direction:ltr;padding:2px 3px 2px 3px;}.ritz .waffle .s0{background-color:#ffffff;text-align:left;color:#666666;font-family:'docs-Roboto',Arial;font-size:10pt;vertical-align:bottom;white-space:nowrap;direction:ltr;padding:2px 3px 2px 3px;}.ritz .waffle .s34{border-bottom:1px SOLID #000000;background-color:#f3f3f3;text-align:center;color:#555555;font-family:'docs-Roboto',Arial;font-size:17pt;vertical-align:middle;white-space:normal;overflow:hidden;direction:ltr;padding:2px 3px 2px 3px;}.ritz .waffle .s16{border-bottom:1px SOLID #000000;border-right:1px SOLID #000000;background-color:#a8ffd5;text-align:center;color:#555555;font-family:'docs-Roboto',Arial;font-size:15pt;vertical-align:middle;white-space:normal;overflow:hidden;direction:ltr;padding:2px 3px 2px 3px;}.ritz .waffle .s45{background-color:#ffffff;text-align:right;color:#555555;font-family:'docs-Roboto',Arial;font-size:10pt;vertical-align:middle;white-space:nowrap;direction:ltr;padding:2px 3px 2px 3px;}.ritz .waffle .s39{border-bottom:1px SOLID #000000;background-color:#ffffff;text-align:left;color:#555555;font-family:'docs-Roboto',Arial;font-size:10pt;vertical-align:middle;white-space:normal;overflow:hidden;direction:ltr;padding:2px 3px 2px 3px;}.ritz .waffle .s13{border-bottom:1px SOLID #000000;border-right:1px SOLID #000000;background-color:#cfe2f3;text-align:center;color:#555555;font-family:'docs-Roboto',Arial;font-size:16pt;vertical-align:middle;white-space:normal;overflow:hidden;direction:ltr;padding:2px 3px 2px 3px;}.ritz .waffle .s42{border-bottom:1px SOLID #000000;border-right:1px SOLID #d9d9d9;background-color:#f3f3f3;text-align:right;color:#434343;font-family:'docs-Roboto',Arial;font-size:10pt;vertical-align:middle;white-space:normal;overflow:hidden;direction:ltr;padding:2px 3px 2px 3px;}.ritz .waffle .s2{background-color:#ffffff;text-align:left;color:#434343;font-family:'docs-Roboto',Arial;font-size:10pt;vertical-align:bottom;white-space:nowrap;direction:ltr;padding:2px 3px 2px 3px;}.ritz .waffle .s32{border-right:1px SOLID #000000;background-color:#f3f3f3;text-align:center;color:#555555;font-family:'docs-Roboto',Arial;font-size:17pt;vertical-align:middle;white-space:normal;overflow:hidden;direction:ltr;padding:2px 3px 2px 3px;}.ritz .waffle .s12{border-bottom:1px SOLID #f3f3f3;border-right:1px SOLID #000000;background-color:#f3f3f3;text-align:right;color:#434343;font-family:'docs-Roboto',Arial;font-size:10pt;vertical-align:middle;white-space:normal;overflow:hidden;direction:ltr;padding:2px 3px 2px 3px;}.ritz .waffle .s1{border-bottom:1px SOLID #d9d9d9;background-color:#ffffff;text-align:center;font-weight:bold;color:#d9d9d9;font-family:'docs-Roboto',Arial;font-size:19pt;vertical-align:bottom;white-space:nowrap;direction:ltr;padding:2px 3px 2px 3px;}.ritz .waffle .s10{border-bottom:1px SOLID #000000;border-right:1px SOLID #000000;background-color:#f3f3f3;text-align:center;font-weight:bold;color:#555555;font-family:'docs-Roboto',Arial;font-size:21pt;vertical-align:middle;white-space:normal;overflow:hidden;direction:ltr;padding:2px 3px 2px 3px;}.ritz .waffle .s26{background-color:#ffffff;text-align:center;color:#555555;font-family:'docs-Roboto',Arial;font-size:17pt;vertical-align:middle;white-space:normal;overflow:hidden;direction:ltr;padding:2px 3px 2px 3px;}.ritz .waffle .s5{border-bottom:1px SOLID #000000;border-right:1px SOLID #000000;background-color:#efefef;text-align:center;color:#555555;font-family:'docs-Roboto',Arial;font-size:17pt;vertical-align:middle;white-space:normal;overflow:hidden;direction:ltr;padding:2px 3px 2px 3px;}.ritz .waffle .s44{background-color:#ffffff;text-align:left;color:#000000;font-family:'docs-Roboto',Arial;font-size:10pt;vertical-align:middle;white-space:nowrap;direction:ltr;padding:2px 3px 2px 3px;}.ritz .waffle .s6{border-bottom:1px SOLID #000000;background-color:#ffffff;text-align:center;font-weight:bold;color:#d9d9d9;font-family:'docs-Roboto',Arial;font-size:19pt;vertical-align:bottom;white-space:nowrap;direction:ltr;padding:2px 3px 2px 3px;}.ritz .waffle .s22{border-bottom:1px SOLID #000000;border-right:1px SOLID #000000;background-color:#f3f3f3;text-align:center;color:#555555;font-family:'docs-Roboto',Arial;font-size:17pt;vertical-align:middle;white-space:normal;overflow:hidden;direction:ltr;padding:2px 3px 2px 3px;}.ritz .waffle .s8{border-right:1px SOLID #000000;background-color:#ffffff;text-align:left;color:#000000;font-family:'docs-Roboto',Arial;font-size:10pt;vertical-align:middle;white-space:nowrap;direction:ltr;padding:2px 3px 2px 3px;}.ritz .waffle .s21{border-right:1px SOLID #000000;background-color:#ffffff;text-align:center;color:#555555;font-family:'docs-Roboto',Arial;font-size:17pt;vertical-align:middle;white-space:normal;overflow:hidden;direction:ltr;padding:2px 3px 2px 3px;}.ritz .waffle .s30{background-color:#f3f3f3;text-align:center;color:#555555;font-family:'docs-Roboto',Arial;font-size:17pt;vertical-align:middle;white-space:normal;overflow:hidden;direction:ltr;padding:2px 3px 2px 3px;}.ritz .waffle .s15{border-bottom:1px SOLID #f3f3f3;border-right:1px SOLID #000000;background-color:#ffffff;text-align:right;color:#434343;font-family:'docs-Roboto',Arial;font-size:10pt;vertical-align:middle;white-space:normal;overflow:hidden;direction:ltr;padding:2px 3px 2px 3px;}.ritz .waffle .s24{border-bottom:1px SOLID #000000;border-right:1px SOLID #000000;background-color:#ffffff;text-align:center;color:#555555;font-family:'docs-Roboto',Arial;font-size:17pt;vertical-align:middle;white-space:normal;overflow:hidden;direction:ltr;padding:2px 3px 2px 3px;}</style><div class="ritz grid-container" dir="ltr"><table class="waffle no-grid" cellspacing="0" cellpadding="0"><thead><tr><th class="row-header freezebar-origin-ltr"></th><th id="0C0" style="width:20px;" class="column-headers-background">A</th><th id="0C1" style="width:97px;" class="column-headers-background">B</th><th id="0C2" style="width:94px;" class="column-headers-background">C</th><th id="0C3" style="width:94px;" class="column-headers-background">D</th><th id="0C4" style="width:94px;" class="column-headers-background">E</th><th id="0C5" style="width:94px;" class="column-headers-background">F</th><th id="0C6" style="width:18px;" class="column-headers-background">G</th><th id="0C7" style="width:96px;" class="column-headers-background">H</th><th id="0C8" style="width:102px;" class="column-headers-background">I</th><th id="0C9" style="width:102px;" class="column-headers-background">J</th><th id="0C10" style="width:102px;" class="column-headers-background">K</th><th id="0C11" style="width:102px;" class="column-headers-background">L</th><th id="0C12" style="width:102px;" class="column-headers-background">M</th><th id="0C13" style="width:102px;" class="column-headers-background">N</th><th id="0C14" style="width:102px;" class="column-headers-background">O</th><th id="0C15" style="width:102px;" class="column-headers-background">P</th><th id="0C16" style="width:144px;" class="column-headers-background">Q</th><th id="0C17" style="width:18px;" class="column-headers-background">R</th><th id="0C18" style="width:96px;" class="column-headers-background">S</th><th id="0C19" style="width:102px;" class="column-headers-background">T</th><th id="0C20" style="width:102px;" class="column-headers-background">U</th><th id="0C21" style="width:102px;" class="column-headers-background">V</th><th id="0C22" style="width:102px;" class="column-headers-background">W</th><th id="0C23" style="width:102px;" class="column-headers-background">X</th><th id="0C24" style="width:144px;" class="column-headers-background">Y</th></tr></thead><tbody><tr style="height: 10px"><th id="0R0" style="height: 10px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 10px">1</div></th><td class="s0"></td><td class="s1" dir="ltr" colspan="5" rowspan="2">Vendredi 16 Février</td><td class="s0"></td><td class="s1" dir="ltr" colspan="10" rowspan="2">Samedi 17 Février</td><td class="s0"></td><td class="s1" dir="ltr" colspan="7" rowspan="2">Dimanche 18 Février</td></tr><tr style="height: 29px"><th id="0R1" style="height: 29px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 29px">2</div></th><td class="s2"></td><td class="s2"></td><td class="s2"></td></tr><tr style="height: 29px"><th id="0R2" style="height: 29px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 29px">3</div></th><td class="s2"></td><td class="s3" dir="ltr"></td><td class="s3" dir="ltr"></td><td class="s3" dir="ltr"></td><td class="s3" dir="ltr"></td><td class="s3" dir="ltr"></td><td class="s2"></td><td class="s4" dir="ltr"></td><td class="s5" dir="ltr" colspan="2">Valorant</td><td class="s5" dir="ltr" colspan="2">CS2</td><td class="s5" dir="ltr" colspan="2">TM</td><td class="s5" dir="ltr" colspan="2">LoL</td><td class="s5" dir="ltr" rowspan="2">Public</td><td class="s2"></td><td class="s4" dir="ltr"></td><td class="s5" dir="ltr" rowspan="2">Valorant</td><td class="s5" dir="ltr" colspan="2">CS2</td><td class="s5" dir="ltr" rowspan="2">TM</td><td class="s5" dir="ltr" rowspan="2">LoL</td><td class="s5" dir="ltr" rowspan="2">Public</td></tr><tr style="height: 29px"><th id="0R3" style="height: 29px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 29px">4</div></th><td class="s2"></td><td class="s6" dir="ltr"></td><td class="s6" dir="ltr"></td><td class="s6" dir="ltr"></td><td class="s6" dir="ltr"></td><td class="s6" dir="ltr"></td><td class="s2"></td><td class="s7" dir="ltr"></td><td class="s5" dir="ltr">Winner</td><td class="s5" dir="ltr">Looser</td><td class="s5" dir="ltr">Winner</td><td class="s5" dir="ltr">Looser</td><td class="s5" dir="ltr">Winner</td><td class="s5" dir="ltr">Looser</td><td class="s5" dir="ltr">Winner</td><td class="s5" dir="ltr">Looser</td><td class="s2"></td><td class="s7" dir="ltr"></td><td class="s5" dir="ltr">Winner</td><td class="s5" dir="ltr">Looser</td></tr><tr style="height: 29px"><th id="0R4" style="height: 29px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 29px">5</div></th><td class="s8"></td><td class="s9" dir="ltr">08:00 </td><td class="s10" dir="ltr" colspan="4" rowspan="24">Fermé au public </td><td class="s11"></td><td class="s12" dir="ltr">08:00 </td><td class="s13" dir="ltr" colspan="9">Ouverture Halle</td><td class="s11"></td><td class="s12" dir="ltr">08:00 </td><td class="s13" dir="ltr" colspan="6">Ouverture Halle</td></tr><tr style="height: 29px"><th id="0R5" style="height: 29px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 29px">6</div></th><td class="s8"></td><td class="s14" dir="ltr">08:30 </td><td class="s11"></td><td class="s15" dir="ltr">08:30 </td><td class="s16" dir="ltr" colspan="9" rowspan="2">Début des tournois</td><td class="s11"></td><td class="s15" dir="ltr">08:30 </td><td class="s16" dir="ltr" colspan="6">Reprise des tournois</td></tr><tr style="height: 29px"><th id="0R6" style="height: 29px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 29px">7</div></th><td class="s8"></td><td class="s9" dir="ltr">09:00 </td><td class="s11"></td><td class="s12" dir="ltr">09:00 </td><td class="s11"></td><td class="s12" dir="ltr">09:00 </td><td class="s17" dir="ltr" rowspan="6">Finale looser BO3</td><td class="s18" dir="ltr" rowspan="7">1/2 BO3</td><td class="s18" dir="ltr" rowspan="3">1/8 BO1</td><td class="s19" dir="ltr" rowspan="3">Finale Amateur</td><td class="s20" dir="ltr" rowspan="7">Looser 4</td><td class="s21" dir="ltr"></td></tr><tr style="height: 29px"><th id="0R7" style="height: 29px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 29px">8</div></th><td class="s8"></td><td class="s14" dir="ltr">09:30 </td><td class="s11"></td><td class="s15" dir="ltr">09:30 </td><td class="s17" dir="ltr" colspan="2" rowspan="3">Match poule 1</td><td class="s18" dir="ltr" colspan="2" rowspan="3">Match 1</td><td class="s19" dir="ltr" colspan="2" rowspan="3">Seeding</td><td class="s20" dir="ltr" colspan="2" rowspan="7">Match Poule</td><td class="s21" dir="ltr"></td><td class="s11"></td><td class="s15" dir="ltr">09:30 </td><td class="s21" dir="ltr"></td></tr><tr style="height: 29px"><th id="0R8" style="height: 29px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 29px">9</div></th><td class="s8"></td><td class="s9" dir="ltr">10:00 </td><td class="s11"></td><td class="s12" dir="ltr">10:00 </td><td class="s21" dir="ltr"></td><td class="s11"></td><td class="s12" dir="ltr">10:00 </td><td class="s21" dir="ltr"></td></tr><tr style="height: 29px"><th id="0R9" style="height: 29px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 29px">10</div></th><td class="s8"></td><td class="s14" dir="ltr">10:30 </td><td class="s11"></td><td class="s15" dir="ltr">10:30 </td><td class="s21" dir="ltr"></td><td class="s11"></td><td class="s15" dir="ltr">10:30 </td><td class="s22" dir="ltr">Pause</td><td class="s19" dir="ltr" rowspan="3">Finale Elite sur scène</td><td class="s21" dir="ltr"></td></tr><tr style="height: 29px"><th id="0R10" style="height: 29px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 29px">11</div></th><td class="s8"></td><td class="s9" dir="ltr">11:00 </td><td class="s11"></td><td class="s12" dir="ltr">11:00 </td><td class="s17" dir="ltr" colspan="2" rowspan="3">Match Poule 2</td><td class="s18" dir="ltr" colspan="2" rowspan="2">Match 2</td><td class="s22" dir="ltr" colspan="2">Pause</td><td class="s21" dir="ltr"></td><td class="s11"></td><td class="s12" dir="ltr">11:00 </td><td class="s18" dir="ltr" rowspan="3">1/6 BO1</td><td class="s21" dir="ltr"></td></tr><tr style="height: 29px"><th id="0R11" style="height: 29px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 29px">12</div></th><td class="s8"></td><td class="s14" dir="ltr">11:30 </td><td class="s11"></td><td class="s15" dir="ltr">11:30 </td><td class="s19" dir="ltr" colspan="2" rowspan="2">Cup 8eme</td><td class="s21" dir="ltr"></td><td class="s11"></td><td class="s15" dir="ltr">11:30 </td><td class="s21" dir="ltr"></td></tr><tr style="height: 29px"><th id="0R12" style="height: 29px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 29px">13</div></th><td class="s8"></td><td class="s9" dir="ltr">12:00 </td><td class="s11"></td><td class="s12" dir="ltr">12:00 </td><td class="s18" dir="ltr" colspan="2" rowspan="2">Match 3</td><td class="s21" dir="ltr"></td><td class="s11"></td><td class="s12" dir="ltr">12:00 </td><td class="s23" dir="ltr" rowspan="2">Installation sur scène</td><td class="s21" dir="ltr"></td><td class="s21" dir="ltr"></td></tr><tr style="height: 29px"><th id="0R13" style="height: 29px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 29px">14</div></th><td class="s8"></td><td class="s14" dir="ltr">12:30 </td><td class="s11"></td><td class="s15" dir="ltr">12:30 </td><td class="s17" dir="ltr" colspan="2" rowspan="2">Match poule 3</td><td class="s22" dir="ltr" colspan="2" rowspan="2">Pause</td><td class="s24" dir="ltr"></td><td class="s11"></td><td class="s15" dir="ltr">12:30 </td><td class="s22" dir="ltr" colspan="2">Pause</td><td class="s21" dir="ltr"></td><td class="s22" dir="ltr" rowspan="2">Pause</td><td class="s21" dir="ltr"></td></tr><tr style="height: 29px"><th id="0R14" style="height: 29px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 29px">15</div></th><td class="s8"></td><td class="s9" dir="ltr">13:00 </td><td class="s11"></td><td class="s12" dir="ltr">13:00 </td><td class="s22" dir="ltr" colspan="2" rowspan="2">Pause</td><td class="s22" dir="ltr" colspan="2" rowspan="2">Pause</td><td class="s25" dir="ltr" rowspan="2">Conférence inclusion dans la Tech</td><td class="s11"></td><td class="s12" dir="ltr">13:00 </td><td class="s17" dir="ltr" rowspan="5">Grande Finale BO3 sur scène</td><td class="s18" dir="ltr" rowspan="6">Finale BO3</td><td class="s18" dir="ltr" rowspan="2">1/4 BO1</td><td class="s21" dir="ltr"></td><td class="s21" dir="ltr"></td></tr><tr style="height: 29px"><th id="0R15" style="height: 29px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 29px">16</div></th><td class="s8"></td><td class="s14" dir="ltr">13:30 </td><td class="s11"></td><td class="s15" dir="ltr">13:30 </td><td class="s17" dir="ltr" colspan="2" rowspan="2">Barrage</td><td class="s19" dir="ltr" rowspan="2">1/4</td><td class="s19" dir="ltr" rowspan="2">1/32</td><td class="s11"></td><td class="s15" dir="ltr">13:30 </td><td class="s26" dir="ltr"></td><td class="s26" dir="ltr"></td><td class="s21" dir="ltr"></td></tr><tr style="height: 29px"><th id="0R16" style="height: 29px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 29px">17</div></th><td class="s8"></td><td class="s9" dir="ltr">14:00 </td><td class="s11"></td><td class="s12" dir="ltr">14:00 </td><td class="s18" dir="ltr" colspan="2" rowspan="2">Match 4</td><td class="s20" dir="ltr" colspan="2" rowspan="4">Match Poule</td><td class="s21" dir="ltr"></td><td class="s11"></td><td class="s12" dir="ltr">14:00 </td><td class="s22" dir="ltr">Pause</td><td class="s26" dir="ltr"></td><td class="s26" dir="ltr"></td><td class="s21" dir="ltr"></td></tr><tr style="height: 29px"><th id="0R17" style="height: 29px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 29px">18</div></th><td class="s8"></td><td class="s14" dir="ltr">14:30 </td><td class="s11"></td><td class="s15" dir="ltr">14:30 </td><td class="s5" dir="ltr" colspan="2" rowspan="2">Pause</td><td class="s22" dir="ltr" rowspan="2">Pause</td><td class="s19" dir="ltr" rowspan="2">1/16</td><td class="s21" dir="ltr"></td><td class="s11"></td><td class="s15" dir="ltr">14:30 </td><td class="s18" dir="ltr" rowspan="2">1/2 BO1</td><td class="s26" dir="ltr"></td><td class="s26" dir="ltr"></td><td class="s21" dir="ltr"></td></tr><tr style="height: 29px"><th id="0R18" style="height: 29px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 29px">19</div></th><td class="s8"></td><td class="s9" dir="ltr">15:00 </td><td class="s11"></td><td class="s12" dir="ltr">15:00 </td><td class="s18" dir="ltr" colspan="2" rowspan="2">Match 5</td><td class="s21" dir="ltr"></td><td class="s11"></td><td class="s12" dir="ltr">15:00 </td><td class="s26" dir="ltr"></td><td></td><td class="s21" dir="ltr"></td></tr><tr style="height: 29px"><th id="0R19" style="height: 29px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 29px">20</div></th><td class="s8"></td><td class="s14" dir="ltr">15:30 </td><td class="s11"></td><td class="s15" dir="ltr">15:30 </td><td class="s17" dir="ltr" colspan="2" rowspan="5">1/4 BO3</td><td class="s19" dir="ltr" rowspan="2">1/2</td><td class="s19" dir="ltr" rowspan="2">1/8</td><td class="s21" dir="ltr"></td><td class="s11"></td><td class="s27" dir="ltr">15:30 </td><td class="s28"></td><td class="s22" dir="ltr" rowspan="2">Pause</td><td class="s26" dir="ltr"></td><td></td><td class="s21" dir="ltr"></td></tr><tr style="height: 29px"><th id="0R20" style="height: 29px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 29px">21</div></th><td class="s8"></td><td class="s9" dir="ltr">16:00 </td><td class="s11"></td><td class="s12" dir="ltr">16:00 </td><td class="s22" dir="ltr" colspan="2">Fin Poules</td><td class="s22" dir="ltr" colspan="2">Pause</td><td class="s21" dir="ltr"></td><td class="s11"></td><td class="s29" dir="ltr">16:00 </td><td class="s21" dir="ltr"></td><td class="s30" dir="ltr"></td><td class="s26" dir="ltr"></td><td class="s31"></td><td class="s21" dir="ltr"></td></tr><tr style="height: 29px"><th id="0R21" style="height: 29px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 29px">22</div></th><td class="s8"></td><td class="s14" dir="ltr">16:30 </td><td class="s11"></td><td class="s15" dir="ltr">16:30 </td><td class="s18" dir="ltr" rowspan="6">1/6 BO3 Winner</td><td class="s21" dir="ltr"></td><td class="s22" dir="ltr" rowspan="2">Pause</td><td class="s19" dir="ltr" rowspan="2">1/4</td><td class="s20" dir="ltr" colspan="2" rowspan="2">1/4</td><td class="s21" dir="ltr"></td><td class="s11"></td><td class="s27" dir="ltr">16:30 </td><td class="s21" dir="ltr"></td><td class="s32" dir="ltr" rowspan="3">Pause</td><td class="s18" dir="ltr" rowspan="2">Finale BO1</td><td class="s21" dir="ltr"></td><td class="s20" dir="ltr" rowspan="7">Finale BO5 sur scène</td><td class="s21" dir="ltr"></td></tr><tr style="height: 29px"><th id="0R22" style="height: 29px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 29px">23</div></th><td class="s8"></td><td class="s9" dir="ltr">17:00 </td><td class="s11"></td><td class="s12" dir="ltr">17:00 </td><td class="s21" dir="ltr"></td><td class="s21" dir="ltr"></td><td class="s11"></td><td class="s9" dir="ltr">17:00 </td><td class="s21" dir="ltr"></td><td class="s21" dir="ltr"></td><td class="s21" dir="ltr"></td></tr><tr style="height: 29px"><th id="0R23" style="height: 29px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 29px">24</div></th><td class="s8"></td><td class="s14" dir="ltr">17:30 </td><td class="s11"></td><td class="s15" dir="ltr">17:30 </td><td class="s21" dir="ltr"></td><td class="s19" dir="ltr" rowspan="2">Finale winner</td><td class="s19" dir="ltr" rowspan="2">1/2</td><td class="s20" dir="ltr" rowspan="6">1/2</td><td class="s26" dir="ltr"></td><td class="s21" dir="ltr"></td><td class="s11"></td><td class="s14" dir="ltr">17:30 </td><td class="s21" dir="ltr"></td><td class="s32" dir="ltr"></td><td class="s21" dir="ltr"></td><td class="s21" dir="ltr"></td></tr><tr style="height: 29px"><th id="0R24" style="height: 29px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 29px">25</div></th><td class="s8"></td><td class="s9" dir="ltr">18:00 </td><td class="s11"></td><td class="s12" dir="ltr">18:00 </td><td class="s5" dir="ltr" colspan="2" rowspan="2">Pause</td><td class="s21" dir="ltr"></td><td class="s33" dir="ltr"></td><td class="s21" dir="ltr"></td><td class="s11"></td><td class="s9" dir="ltr">18:00 </td><td class="s21" dir="ltr"></td><td class="s34" dir="ltr"></td><td class="s22" dir="ltr"></td><td class="s21" dir="ltr"></td><td class="s21" dir="ltr"></td></tr><tr style="height: 29px"><th id="0R25" style="height: 29px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 29px">26</div></th><td class="s8"></td><td class="s14" dir="ltr">18:30 </td><td class="s11"></td><td class="s15" dir="ltr">18:30 </td><td class="s21" dir="ltr"></td><td class="s19" dir="ltr" colspan="2" rowspan="2">Finale Loser</td><td class="s20" dir="ltr" rowspan="2">Looser 1</td><td class="s21" dir="ltr"></td><td class="s11"></td><td class="s14" dir="ltr">18:30 </td><td class="s21" dir="ltr"></td><td class="s35" dir="ltr" colspan="2">Installation sur scène</td><td class="s21" dir="ltr"></td><td class="s21" dir="ltr"></td></tr><tr style="height: 29px"><th id="0R26" style="height: 29px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 29px">27</div></th><td class="s8"></td><td class="s9" dir="ltr">19:00 </td><td class="s11"></td><td class="s12" dir="ltr">19:00 </td><td class="s17" dir="ltr" rowspan="6">1/2 BO3</td><td class="s17" dir="ltr" rowspan="2">Looser round 1</td><td class="s24" dir="ltr"></td><td class="s21" dir="ltr"></td><td class="s11"></td><td class="s9" dir="ltr">19:00 </td><td class="s26" dir="ltr"></td><td></td><td></td><td class="s21" dir="ltr"></td><td></td></tr><tr style="height: 29px"><th id="0R27" style="height: 29px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 29px">28</div></th><td class="s8"></td><td class="s14" dir="ltr">19:30 </td><td class="s11"></td><td class="s15" dir="ltr">19:30 </td><td class="s22" dir="ltr" colspan="2" rowspan="2">Pause</td><td class="s21" dir="ltr" colspan="2" rowspan="5"></td><td class="s26" dir="ltr"></td><td class="s21" dir="ltr"></td><td class="s11"></td><td class="s14" dir="ltr">19:30 </td><td class="s26" dir="ltr"></td><td></td><td></td><td class="s21" dir="ltr"></td><td></td></tr><tr style="height: 29px"><th id="0R28" style="height: 29px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 29px">29</div></th><td class="s8"></td><td class="s12" dir="ltr">20:00 </td><td class="s36" dir="ltr" colspan="4">Ouverture au Public &amp; Acceuil des joueurs</td><td class="s11"></td><td class="s12" dir="ltr">20:00 </td><td class="s21" dir="ltr"></td><td class="s33" dir="ltr"></td><td class="s21" dir="ltr"></td><td class="s11"></td><td class="s9" dir="ltr">20:00 </td><td class="s26" dir="ltr"></td><td></td><td></td><td class="s26" dir="ltr"></td><td></td><td></td></tr><tr style="height: 29px"><th id="0R29" style="height: 29px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 29px">30</div></th><td class="s8"></td><td class="s14" dir="ltr">20:30 </td><td class="s37" dir="ltr"></td><td class="s37"></td><td class="s37"></td><td class="s38"></td><td class="s11"></td><td class="s15" dir="ltr">20:30 </td><td class="s21" dir="ltr"></td><td class="s18" dir="ltr" rowspan="6">1/4 BO3 Winner</td><td class="s26" dir="ltr"></td><td class="s22" dir="ltr" colspan="2">Pause</td><td class="s21" dir="ltr"></td><td class="s11"></td><td class="s14" dir="ltr">20:30 </td><td class="s26" dir="ltr"></td><td class="s31"></td><td class="s31"></td><td class="s26" dir="ltr"></td><td></td><td></td></tr><tr style="height: 29px"><th id="0R30" style="height: 29px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 29px">31</div></th><td class="s8"></td><td class="s9" dir="ltr">21:00 </td><td class="s39" dir="ltr"></td><td class="s39"></td><td class="s39"></td><td class="s40"></td><td class="s11"></td><td class="s12" dir="ltr">21:00 </td><td class="s21" dir="ltr"></td><td class="s26" dir="ltr"></td><td class="s20" dir="ltr" rowspan="7">Finale Winner</td><td class="s20" dir="ltr" rowspan="2">Looser 2</td><td class="s21" dir="ltr"></td><td class="s11"></td><td class="s9" dir="ltr">21:00 </td><td class="s21" dir="ltr"></td><td class="s18" dir="ltr" colspan="2" rowspan="6">Finale BO3 sur scène</td><td class="s26" dir="ltr"></td><td></td><td></td></tr><tr style="height: 29px"><th id="0R31" style="height: 29px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 29px">32</div></th><td class="s8"></td><td class="s15" dir="ltr">21:30 </td><td class="s41" dir="ltr" colspan="4" rowspan="3">Concert par Pixel Brass Band</td><td class="s11"></td><td class="s15" dir="ltr">21:30 </td><td class="s24" dir="ltr"></td><td class="s26" dir="ltr"></td><td class="s21" dir="ltr"></td><td class="s11"></td><td class="s14" dir="ltr">21:30 </td><td class="s21" dir="ltr"></td><td class="s26" dir="ltr"></td><td></td><td></td></tr><tr style="height: 29px"><th id="0R32" style="height: 29px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 29px">33</div></th><td class="s8"></td><td class="s12" dir="ltr">22:00 </td><td class="s11"></td><td class="s12" dir="ltr">22:00 </td><td class="s17" dir="ltr" rowspan="5">Finale BO3</td><td class="s17" dir="ltr" rowspan="2">Looser round 2</td><td class="s26" dir="ltr"></td><td class="s26" dir="ltr"></td><td class="s21" dir="ltr"></td><td class="s26" dir="ltr"></td><td class="s21" dir="ltr"></td><td class="s11"></td><td class="s9" dir="ltr">22:00 </td><td class="s28"></td><td></td><td></td><td></td></tr><tr style="height: 29px"><th id="0R33" style="height: 29px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 29px">34</div></th><td class="s8"></td><td class="s15" dir="ltr">22:30 </td><td class="s11"></td><td class="s15" dir="ltr">22:30 </td><td class="s26" dir="ltr"></td><td class="s26" dir="ltr"></td><td class="s21" dir="ltr"></td><td class="s26" dir="ltr"></td><td class="s21" dir="ltr"></td><td class="s11"></td><td class="s14" dir="ltr">22:30 </td><td class="s21" dir="ltr"></td><td class="s26" dir="ltr"></td><td></td><td></td></tr><tr style="height: 29px"><th id="0R34" style="height: 29px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 29px">35</div></th><td class="s8"></td><td class="s9" dir="ltr">23:00 </td><td class="s37" dir="ltr"></td><td class="s37"></td><td class="s37"></td><td class="s38"></td><td class="s11"></td><td class="s12" dir="ltr">23:00 </td><td class="s24" dir="ltr"></td><td class="s26" dir="ltr"></td><td class="s26" dir="ltr"></td><td class="s21" dir="ltr"></td><td class="s33" dir="ltr"></td><td class="s21" dir="ltr"></td><td class="s11"></td><td class="s9" dir="ltr">23:00 </td><td class="s21" dir="ltr"></td><td class="s26" dir="ltr"></td><td></td><td></td></tr><tr style="height: 29px"><th id="0R35" style="height: 29px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 29px">36</div></th><td class="s8"></td><td class="s14" dir="ltr">23:30 </td><td class="s37"></td><td class="s37"></td><td class="s37"></td><td class="s38"></td><td class="s11"></td><td class="s15" dir="ltr">23:30 </td><td class="s17" dir="ltr" rowspan="2">Looser round 3</td><td class="s26" dir="ltr"></td><td class="s26" dir="ltr"></td><td class="s26" dir="ltr"></td><td class="s21" dir="ltr"></td><td class="s20" dir="ltr" rowspan="2">Looser 3</td><td class="s21" dir="ltr"></td><td class="s11"></td><td class="s14" dir="ltr">23:30 </td><td class="s21" dir="ltr"></td><td class="s26" dir="ltr"></td><td></td><td></td></tr><tr style="height: 29px"><th id="0R36" style="height: 29px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 29px">37</div></th><td class="s8"></td><td class="s42" dir="ltr">00:00 </td><td class="s39" dir="ltr"></td><td class="s39"></td><td class="s39"></td><td class="s40"></td><td class="s11"></td><td class="s43" dir="ltr">00:00 </td><td class="s33" dir="ltr"></td><td class="s33" dir="ltr"></td><td class="s33" dir="ltr"></td><td class="s24" dir="ltr"></td><td class="s24" dir="ltr"></td><td class="s11"></td><td class="s42" dir="ltr">00:00 </td><td class="s33" dir="ltr"></td><td class="s33" dir="ltr"></td><td class="s33" dir="ltr"></td><td class="s33" dir="ltr"></td><td class="s33" dir="ltr"></td><td class="s24" dir="ltr"></td></tr><tr style="height: 10px"><th id="0R37" style="height: 10px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 10px">38</div></th><td class="s44"></td><td class="s45"></td><td class="s46"></td><td class="s46"></td><td class="s46"></td><td class="s46"></td><td class="s46"></td><td class="s45"></td><td class="s46"></td><td class="s46"></td><td class="s46"></td><td class="s46"></td><td class="s46"></td><td class="s46"></td><td class="s46"></td><td class="s46"></td><td class="s46"></td><td class="s46"></td><td class="s46" dir="ltr">q</td><td class="s46"></td><td class="s46"></td><td class="s46"></td><td class="s46"></td><td class="s46"></td><td class="s46"></td></tr></tbody></table></div>
```

Reply:
Instead of replacing colors, you should put on some classes in events and let the frontend (or user's browser) pick the colors. Hardcoding colors breaks accessibility

Reply:
I agree with that statement but this is truly impossible to add classes to the schedule code. You have an example 2 messages above but there already is auto-generated classes with some css. And the name of theses auto-generated classes depends of the size and position of the gdoc cells as well as their style depending of the cell colour. 
If you want a comparison, the gdoc is looking like the pages in this one : https://docs.google.com/spreadsheets/d/1E3uvh7L5EqcbwwIeF02Jrz_1JPtoPRaR5xaH5sDJy6k/edit?usp=sharing

Reply:
> The problem is that an exported google spreadsheet already contains css with a hard to use format.

Make another tool to strip that down to the barest of formats possible.

An alternative if you REALLy want to hardcode HTML on this: have it use classes that will already make it coloured correctly.

Alternatively, thing I did not think of when we merged the planning thing: this is just introducing code injection to the website from the admin panel. Wtf.

Reply:
> Alternatively, thing I did not think of when we merged the planning thing: this is just introducing code injection to the website from the admin panel. Wtf.
That's true but i has nothing to do with the planning. Contents can be HTML code and can contains code tags.

Reply:
> Make another tool to strip that down to the barest of formats possible.
> 
> An alternative if you REALLy want to hardcode HTML on this: have it use classes that will already make it coloured correctly.
The solutions you describe are not feasible. The classes already present contain the document format itself and cannot be removed without loss of information. Making a tool that replaces the classes with their respective styles is, in my opinion, a waste of time. 
As for putting colors in dedicated classes, this would require a lot of work just to transfer styles from classes to others, while requiring more hardcoded operations to reduce the classes.
Once again, if you have a solution, i'm listening but this is not as easy as you wrote it. The export

Reply:
> Alternatively, thing I did not think of when we merged the planning thing: this is just introducing code injection to the website from the admin panel. Wtf.

All the CMS is a big code injection module. We assumed we'd have other problems if an user succeeds to login to the admin panel in our threat model.